sdk/src/short_vec.rs

```diff
@@ -179,8 +179,8 @@ impl<'de, T: Deserialize<'de>> Deserialize<'de> for ShortVec<T> {
 /// Return the decoded value and how many bytes it consumed.
 pub fn decode_len(bytes: &[u8]) -> Result<(usize, usize), Box<bincode::ErrorKind>> {
     let short_len: ShortU16 = bincode::deserialize(bytes)?;
-    let num_bytes = bincode::serialized_size(&short_len).unwrap() as usize;
```

Comment:
A test here would be nice. That error looks unreachable.

Reply:
Same as before but return the error instead of panicking even if it is unreachable today